src/main/java/org/springframework/data/couchbase/core/ReactiveUpsertByIdOperationSupport.java

```diff
@@ -71,10 +73,10 @@ public ReactiveUpsertByIdOperationSupport(final ReactiveCouchbaseTemplate templa
 		public Mono<T> one(T object) {
 			return Mono.just(object).flatMap(o -> {
 				CouchbaseDocument converted = template.support().encodeEntity(o);
+				Object updatedObject = template.support().applyUpdatedId(o, converted.getId());
 				return template.getCollection(collection).reactive()
 						.upsert(converted.getId(), converted.export(), buildUpsertOptions()).map(result -> {
-							template.support().applyUpdatedCas(object, result.cas());
```

Comment:
would it make sense to move both apply methods into the upsert map callback so they are done in   one spot?

Reply:
For both id and cas updates, they are done as soon as the value is available. 
There is no efficiency benefit to doing them at the same time even for immutable entities - each update will require that a new object be created.

Reply:
After a lot of thinking, I made this change.